src/simulator/views/map/data/map_data.py

```diff
@@ -40,22 +49,12 @@ def name(self) -> str:
     def name(self) -> NodePath:
         return self.__name
 
-    def _dynamic_colour_callback(self, colour: DynamicColour) -> None:
-        self._services.ev_manager.post(KeyFrameEvent(refresh=True))
-
-    def add_colour(self, name: str, default_colour: Colour, default_visible: bool = True, invoke_callback: bool = True, callback: Optional[Callable[[DynamicColour], None]] = None) -> DynamicColour:
+    def _add_colour(self, name: str, default_colour: Colour, default_visible: bool = True, invoke_callback: bool = True, callback: Optional[Callable[[DynamicColour], None]] = None) -> DynamicColour:
```

Comment:
Why has this method name been changed, is there a particular reason?

Reply:
We don't want anything outside of `MapData` and derived from accessing this method. Adding custom colours (on a per-algorithm basis) should be done by directly accessing `PersistentState` instead of this way.